.github/workflows/unit-tests.yml

```diff
@@ -13,70 +13,72 @@ concurrency:
   cancel-in-progress: true
 
 env:
-  BOOST_VERSION: 1.69.0
+  CMAKE_VERSION: 3.26.0
+  BOOST_VERSION: 1.81.0
   ADDITIONAL_CMAKE_FLAGS: -DRTTR_ENABLE_BENCHMARKS=ON
 
 jobs:
   Linux:
     strategy:
       matrix:
         include:
-          - { compiler: gcc-6,    os: ubuntu-18.04, buildType: Release }
-          - { compiler: gcc-6,    os: ubuntu-18.04, buildType: Debug, coverage: true }
-          - { compiler: clang-7,  os: ubuntu-18.04, buildType: Release }
-          - { compiler: clang-7,  os: ubuntu-18.04, buildType: Debug }
-          - { compiler: clang  ,  os: macos-10.15,  buildType: Debug, boostVersion: 1.74.0 } # Multiple bugs with recent OSX until 1.74
-          # GCC 9 is known to show a few warnings that GCC 10 has "fixed", make sure this doesn't happen for us
```

Comment:
Is GCC-9 really not available on any of the runner images? This entry was especially to avoid a regression on GCC 9 (hence the comment)   
Also the current change means that our supported minimum  GCC version changes from 6 to 10 which is a lot IMO.

It looks like GCC-7 is available in the [Ubuntu 20 repos](https://packages.ubuntu.com/focal/allpackages?format=txt.gz)

So I'd use GCC-7 as the minimum and keep the GCC-9 entry

Reply:
gcc-9 is also so outdated, lets skip those old compilers without c++14 support (I'll update the whole toolchain)

Reply:
What doesn't it support? And I wouldn't say it is outdated: It is the default for Ubuntu 20